streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -271,11 +274,17 @@ public void close(final Map<TopicPartition, Long> ackedOffsets) throws Processor
                     log.error("Failed to close state store {}: ", store.name(), e);
                 }
             }
+            stores.clear();
+        }
 
-            if (ackedOffsets != null) {
-                checkpoint(ackedOffsets);
+        if (!clean && this.eosEnabled && checkpoint != null) {
```

Comment:
nit: remove `this`

Reply:
To make the semantics clearer, I am wondering if we should use two nested `if`:

```
if(eosEnabled && !clean) {
  try {
    if (checkpoint != null) { ... }
  } catch() {...{
}
```

This make it clear, that it's a EOS condition (first var to be checked) and for the EOS, do something if `!clean`.

The `checkpoint != null` if just a guard against NPE and has nothing to do with the actual logic